core/sql/optimizer/OptPhysRelExpr.cpp

```diff
@@ -11814,10 +11814,23 @@ void SortGroupBy::addArrangementAndOrderRequirements(
   {
     // Shouldn't/Can't add a sort order type requirement
     // if we are in DP2
-    if (rg.getStartRequirements()->executeInDP2())
-      rg.addArrangement(groupExpr(),NO_SOT);
-    else
-      rg.addArrangement(groupExpr(),ESP_SOT);
+
+   if( NOT extraOrderExpr().isEmpty())
+   {
+     ValueIdList groupExprCpy(groupExpr());
+     for (ValueId vid = extraOrderExpr().init(); extraOrderExpr().next(vid);
```

Comment:
At this point, extraOrderExpr contains (test_score) only. The groupExpr is required by the SortGroupby, here we just add the "extra " sort key, groupExpr is required to do the sort groupby.
PIVOT_GROUP will force the optimizer to use sort groupby instead of hash groupby. As you can see, above code is in fact in SortGroupBy method. It is in the rule's match() method.
I ignore partial groupby and shortcut groupby. That must be an important issue, one possibility is to prevent them just as we prevent in hash groupby. This is a very critical question, thanks Suresh to remind me this. I need to study these two groupby methods.